Src/AutoNSubstituteUnitTest/SubstituteRelayTest.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using NSubstitute;
+using NSubstitute.Exceptions;
+using Ploeh.AutoFixture.Kernel;
+using Ploeh.TestTypeFoundation;
+using Xunit;
+using Xunit.Extensions;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute.UnitTest
+{
+    public class SubstituteRelayTest
+    {
+        [Fact]
+        public void ClassImplementsISpecimenBuilderToServeAsResidueCollector()
+        {
+            // Fixture setup
+            // Exercise system
+            // Verify outcome
+            Assert.True(typeof(ISpecimenBuilder).IsAssignableFrom(typeof(SubstituteRelay)));
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateThrowsArgumentNullExceptionWhenContextIsNullBecauseItsRequired()
+        {
+            // Fixture setup
+            var sut = new SubstituteRelay();
+            var request = new object();
+            // Exercise system
+            var e = Assert.Throws<ArgumentNullException>(() => sut.Create(request, null));
+            // Verify outcome
+            Assert.Equal("context", e.ParamName);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNotAType()
+        {
+            // Fixture setup
+            var sut = new SubstituteRelay();
+            object request = "beer";
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object result = sut.Create(request, context);
+            // Verify outcome
+            var noSpecimen = Assert.IsType<NoSpecimen>(result);
+            Assert.Same(request, noSpecimen.Request);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestedTypeIsNotAbstract()
+        {
+            // Fixture setup
+            var sut = new SubstituteRelay();
+            object request = typeof(string);
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object result = sut.Create(request, context);
+            // Verify outcome
+            var noSpecimen = Assert.IsType<NoSpecimen>(result);
+            Assert.Same(request, noSpecimen.Request);
+            // Teardown
+        }
+
+        [Theory]
+        [InlineData(typeof(AbstractType))]
+        [InlineData(typeof(IInterface))]
+        public void CreateReturnsObjectResolvedFromContextWhenRequestedTypeIsAbstractOrInterface(Type requestedType)
+        {
+            // Fixture setup
+            var sut = new SubstituteRelay();
+            object request = requestedType;
+            object substitute = Substitute.For(new Type[] { requestedType }, new object[0]);
+            var context = Substitute.For<ISpecimenContext>();
+            context.Resolve(Arg.Any<object>()).Returns(substitute);

```

Comment:
Why would you need `Resolve(Arg.Is<Type>(t => t == requestedType), ...)`? Wouldn't it be enough to write `Resolve(requestedType, ...)`?